### ISSUE TITLE
Test failures when run tests page locally in a browser

### ISSUE THREAD

**Issue by olegabr** [qcubed/framework#862]
Surprisingly for me, tests pass for almost all PHP versions (https://github.com/qcubed/framework/pull/859). But I have failures when run tests page locally in a browser: 
```
ModelConnectorTests > testReference
Pass: Expected true, got [Boolean: true]
Pass: 7
Pass: Forward reference saved correctly through connector.
Pass: Reloaded forward reference connector
Fail: ModelConnectorTests -> Expected true, got [Boolean: false] at [/home/olegabr/work/chess.com/vendor/qcubed/framework/includes/tests/qcubed-unit/ModelConnectorTests.php line 71]
Pass: Forward reference saved correctly through connector.
Pass: Equal expectation [String: New York]

ModelConnectorTests > testOverrides
Fail: ModelConnectorTests -> Street Label was removed by override. at [/home/olegabr/work/chess.com/vendor/qcubed/framework/includes/tests/qcubed-unit/ModelConnectorTests.php line 192]
Fail: ModelConnectorTests -> Equal expectation fails as [NULL] does not match [Integer: 100] at [/home/olegabr/work/chess.com/vendor/qcubed/framework/includes/tests/qcubed-unit/ModelConnectorTests.php line 194]
Fail: ModelConnectorTests -> Equal expectation fails because [Integer: 1] differs from [Integer: 3] by 2 at [/home/olegabr/work/chess.com/vendor/qcubed/framework/includes/tests/qcubed-unit/ModelConnectorTests.php line 198]
Fail: ModelConnectorTests -> Equal expectation fails at character 0 with [People As Team Member] and [Team Members] at [/home/olegabr/work/chess.com/vendor/qcubed/framework/includes/tests/qcubed-unit/ModelConnectorTests.php line 199]
Fail: ModelConnectorTests -> Reverse reference was marked required by override file. at [/home/olegabr/work/chess.com/vendor/qcubed/framework/includes/tests/qcubed-unit/ModelConnectorTests.php line 203]
Fail: ModelConnectorTests -> Required value was detected by list control. at [/home/olegabr/work/chess.com/vendor/qcubed/framework/includes/tests/qcubed-unit/ModelConnectorTests.php line 206]

13/13 test cases complete: 739 passes, 7 fails and 0 exceptions.
```

There should be some difference in cli and web test launchers.

**Comment by olegabr** [comment]
The reason is that the file `/travis/codegen_options.json` is preconfigured, but the default one is not.

**Comment by olegabr** [comment]
If I copy the pre-configured one to the `/project/includes/configuration/` directory and re-codegen, all tests pass. Why not it be there by default?

**Comment by olegabr** [comment]
we can release 3.0 without it. target it to be v3.1

**Comment by spekary** [comment]
This seems like a bug. It should work fine without a codegen_options.json file.

**Comment by olegabr** [comment]
No, it should not. There is a test, for example, that check a type of control generated to be QAutocomplete. But only plain html output is generated without proper `codegen_options.json` file

**Comment by spekary** [comment]
Oh, I see what you are saying. The unit tests fail without it. Thorny issue. Probably the unit test should check to make sure the right file is there before running.

**Comment by spekary** [comment]
Fixed in #1021. Basically, I am just documenting how to set up the tests so that they can run locally.
